setup.py

```diff
@@ -0,0 +1,70 @@
+import os
+import time
+
+from setuptools import setup
+from setuptools_scm.version import guess_next_version
```

Comment:
I see `setup.py` usage for `setuptools_scm` is deprecated: https://github.com/pypa/setuptools_scm#setuppy-usage-deprecated 

It seems like they're leaning towards `pyproject.toml`. I don't have a lot of experience with pipenv but it seems like it could handle the `pyproject.toml` file. If we don't need to move to a `setup.py` configuration and we can stick to `pyproject.toml` I think we should. If just because more and more tools are moving in that direction.

Reply:
Yes, they are leaning that way for sure. I've just never gotten this kind of "auto releasing" scheme to work with a toml configuration. If you know how to do it, let's substituted it. It's just beyond my current knowledge.